### PR TITLE
[AMORO-2188] Fix NPE when only using external optimizers

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

fix:#2188

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by wangtaohz — CHANGES_REQUESTED** [review]
Thanks for your work! 👍 

I tested the fixed code locally, and verified that even without configuring any containers, we can still choose an external container.
image


However, I believe there is no need to modify the config file. After modifying it, this PR can be merged.

File: dist/src/main/arctic-bin/conf/config.yaml [hunk 1]
```diff
@@ -91,11 +91,11 @@ ams:
 #    cluster-name: default
 #    zookeeper-address: 127.0.0.1:2181,127.0.0.1:2182,127.0.0.1:2183
 
-containers:
-  - name: localContainer
-    container-impl: com.netease.arctic.server.manager.LocalOptimizerContainer
-    properties:
-      export.JAVA_HOME: "/opt/java"   # JDK environment
+#containers:
```
[the remark this answers is not in the document]
Comment (tcodehuber) [reply dist/src/main/arctic-bin/conf/config.yaml]: > This `dist/src/main/arctic-bin/conf/config.yaml` is the default configuration file for users when they run a demo of AMS. I believe it is reasonable to keep localContainer, and there is no need to modify it in this PR.

done.

File: dist/src/main/arctic-bin/conf/config.yaml [hunk 2]
```diff
@@ -91,11 +91,11 @@ ams:
...
+#  - name: localContainer
+#    container-impl: com.netease.arctic.server.manager.LocalOptimizerContainer
+#    properties:
+#      export.JAVA_HOME: "/opt/java"   # JDK environment
```
[the remark this answers is not in the document]
Comment (tcodehuber) [reply dist/src/main/arctic-bin/conf/config.yaml]: > What is the reason for removing local container from the default config.yaml?

reverted the unnecessary changes

**Comment by tcodehuber** [comment]
> Thanks for your work! 👍
> 
> I tested the fixed code locally, and verified that even without configuring any containers, we can still choose an external container. image
> 
> However, I believe there is no need to modify the config file. After modifying it, this PR can be merged.

done

**Review by wangtaohz — APPROVED** [review]
LGTM

**Pull request merged**